sklearn/impute/_base.py

```diff
@@ -251,7 +217,7 @@ def _validate_input(self, X):
 
         try:
             X = check_array(X, accept_sparse='csc', dtype=dtype,
-                            force_all_finite=force_all_finite, copy=self.copy)
+                            force_all_finite=force_all_finite, copy=self._copy)
```

Comment:
Is defining `self._copy` needed?

If `add_indicator=True` and `self.copy=False`, the imputer will do the operations in place, and later when `hstack` is called, a copy will be made.

Reply:
We are implementing: fit_indicator -> impute -> transform_concat_indicator.
We can only transform if the operation was not inplace (`copy=True`). If `copy=False`, then we need to have something like: fit_indicator -> transform_indicator -> impute -> concat_indicator.

Then, we could ask ourself if this is necessary since transform and concat will be single line call. The only benefit there will be that the call will be more explicit than just a `transform` and `hstack`.

In the latter case, this is not obvious that we need it since this is a single line for transform, a single line for concat.